pkg/block/fetcher.go

```diff
@@ -533,6 +534,23 @@ func contains(s1 []ulid.ULID, s2 []ulid.ULID) bool {
 	return true
 }
 
+type ReplicaLabelsFilter struct {
+	ReplicaLabels []string
+}
+
+func (f *ReplicaLabelsFilter) Filter(metas map[ulid.ULID]*metadata.Meta, synced GaugeLabeled, view bool) {
+	for u, meta := range metas {
+		labels := meta.Thanos.Labels
+		for _, replicaLabel := range f.ReplicaLabels {
+			if _, exists := labels[replicaLabel]; exists {
+				delete(labels, replicaLabel)
```

Comment:
We need comment that we modify external lable of existing block, constructing fake meta.json. This might be surprising so we need to be extra careful